sql/src/main/java/io/crate/analyze/validator/GroupBySymbolValidator.java

```diff
@@ -53,6 +57,7 @@ private static void validateDataType(Symbol symbol) {
         public Void visitFunction(Function symbol, Void context) {
             switch (symbol.info().type()) {
                 case SCALAR:
+                    validateArguments(symbol.arguments());
```

Comment:
I think for this to also work for nested scalars this would have to traverse the arguments:

```
for (Symbol arg : symbol.arguments()) {
    process(arg, context);
}
```